sessionresolver/repos/cache/cacheRepo.go

```diff
@@ -35,24 +36,27 @@ func (r cacheRepo) InsertOrUpdate(ctx context.Context, id string, obj interface{
 	return r.cacheSetter.InsertOrUpdate(ctx, r.sessionCollectionName, id, r.version, obj)
 }
 
-func (r cacheRepo) GetCacheVersions(ctx context.Context, now time.Time) (map[string]string, error) {
-	versions, err := r.cacheGetter.GetLatestVersions(ctx)
+func (r cacheRepo) GetCacheVersions(ctx context.Context, now time.Time, filterAction string) (map[string]string, error) {
+	cacheCollections, err := r.cacheGetter.GetLatestVersions(ctx)
 	if err != nil {
 		return nil, err
 	}
 	result := make(map[string]string)
 
-	for i := range versions {
+	for _, cacheCollection := range cacheCollections {
+		if filterAction != "" && !slices.IsStrExist(cacheCollection.LockVersionUpon, filterAction) {
+			continue
+		}
 		var latestVersion cacheStorage.Version
-		for _, v := range versions[i].Versions {
+		for _, v := range cacheCollection.Versions {
 			if (latestVersion.TimedTo.IsZero() || v.TimedTo.After(latestVersion.TimedTo)) && v.TimedTo.Before(now) {
```

Comment:
latestVersion.TimedTo.IsZero()  ?
where is the expiry after end of shift ?

Reply:
latestVersion.TimedTo.IsZero() means latestVersion is empty

![image](https://github.com/orchestd/session/assets/126398177/a3c62118-3d99-4582-b5d3-a74fe256757f)